### PR TITLE
Change back Illinois Courts Rule 23 status to Unpublished

### PR DESCRIPTION
**Pull request by satsuki-chan**
**Rule 23** should remain non-precedential, Unpublished.

Fixes: #405

### CODE REVIEW

**Comment by mlissner** [comment]
@flooie can you review this one, please?

File: juriscraper/opinions/united_states/state/ill.py [hunk 1]
```diff
@@ -6,8 +6,6 @@
    2014-12-02: Updated by Mike Lissner to remove the summaries code.
    2016-02-26: Updated by arderyp: simplified thanks to new id attribute identifying decisions table
    2016-03-27: Updated by arderyp: fixed to handled non-standard formatting
-   2021-11-02: Status validation based on Amendment to Rule 23 rulings
-               https://chicagocouncil.org/illinois-supreme-court-amendment-to-rule-23/
 """
```
Comment (flooie) [inline juriscraper/opinions/united_states/state/ill.py:9]: Also -  you should follow the pattern we have above for date updated - by whom and what was changed.

**Comment by flooie** [comment]
@satsuki-chan this is a good start and probably only a couple tweaks to wrap it up.  Please add a note in both files that were changed dating and explaining the update.

**Review by flooie — CHANGES_REQUESTED** [review]

File: juriscraper/opinions/united_states/state/illappct.py [hunk 2]
```diff
@@ -14,9 +14,7 @@ class Site(ill.Site):
     def __init__(self, *args, **kwargs):
         super(Site, self).__init__(*args, **kwargs)
         self.court_id = self.__module__
-        self.url = (
-            "https://www.illinoiscourts.gov/top-level-opinions?type=appellate"
-        )
+        self.url = "https://www.illinoiscourts.gov/top-level-opinions?type=all%20appellate"
```
[the remark this answers is not in the document]
Comment (satsuki-chan) [reply juriscraper/opinions/united_states/state/illappct.py]: Changed URL

File: juriscraper/opinions/united_states/state/ill.py (hunk 1 above)
Comment (satsuki-chan) [reply juriscraper/opinions/united_states/state/ill.py]: Added comment with change to both scrapers.

File: juriscraper/opinions/united_states/state/ill.py [hunk 3]
```diff
@@ -39,16 +35,11 @@ def _process_html(self):
...
+                decision_type = get_row_column_text(row, 5)
+                if decision_type == "Rule 23":
+                    status = "Unpublished"
                 else:
-                    decision_type = get_row_column_text(row, 5)
-                    if decision_type == "Rule 23":
-                        status = "Unpublished"
-                    else:
-                        status = "Published"
+                    status = "Published"
```
[the remark this answers is not in the document]
Comment (satsuki-chan) [reply juriscraper/opinions/united_states/state/ill.py]: Changed validation based on the '-U' (or '-U[X: any char]') ending of the citation.

File: juriscraper/opinions/united_states/state/ill.py [hunk 4]
```diff
@@ -1,13 +1,13 @@
...
-   2016-03-27: Updated by arderyp: fixed to handled non-standard formatting
-   2021-11-02: Status validation based on Amendment to Rule 23 rulings
-               https://chicagocouncil.org/illinois-supreme-court-amendment-to-rule-23/
+History:
+  2013-08-16: Created by Krist Jin
+  2014-12-02: Updated by Mike Lissner to remove the summaries code.
+  2016-02-26: Updated by arderyp: simplified thanks to new id attribute identifying decisions table
+  2016-03-27: Updated by arderyp: fixed to handled non-standard formatting
+  2021-11-02: Updated by satsuki-chan: Updated to new page design and updated status validation based on Amendment to Rule 23 rulings
+              https://chicagocouncil.org/illinois-supreme-court-amendment-to-rule-23/
```
Comment (flooie) [inline juriscraper/opinions/united_states/state/ill.py:10]: I would just drop everything after 'and'.

File: juriscraper/opinions/united_states/state/illappct.py [hunk 5]
```diff
@@ -1,8 +1,11 @@
...
 """
-# Author: Krist Jin
-# 2013-08-18: Created.
-# 2014-07-17: Updated by mlr to remedy InsanityException.
+Author: Krist Jin
+History:
+  2013-08-18: Created.
+  2014-07-17: Updated by mlr to remedy InsanityException.
+  2021-11-02: Updated by satsuki-chan: Updated to new page design and updated status validation based on Amendment to Rule 23 rulings
+              https://chicagocouncil.org/illinois-supreme-court-amendment-to-rule-23/
```
Comment (flooie) [inline juriscraper/opinions/united_states/state/illappct.py:8]: Same as above.

File: juriscraper/opinions/united_states/state/illappct.py [hunk 6]
```diff
@@ -19,7 +22,7 @@ def __init__(self, *args, **kwargs):
         )
 
     def extract_docket(self, case_citation):
-        # RegEx: "[{YYYY: year_4_digit}|] IL [App|] ({d: appellate district}[st|d|th]) {docket_number}[-U|]"
+        # RegEx: "[{YYYY: year_4_digit}|] IL [App|] ({d: appellate district}[st|d|th]) {docket_number}[|-U|-B|-C|-UB|-UC]"
         search = re.search(r"(?:.+\()(\d+)(?:.+\)\s+)(\d+)", case_citation)
```
Comment (flooie) [inline juriscraper/opinions/united_states/state/illappct.py:26]: Same as above.

**Review by flooie — CHANGES_REQUESTED** [review]
@satsuki-chan - Just because we know that they are using versioning doesn't mean to me that we should drop those letters.  They are informative to us and probably to attorneys out there.  Otherwise I think this is close.  

Also - if I'm reading this code right - it would truncate the citations without the Letter?  If that is the case why did it not fail the tests?

File: juriscraper/opinions/united_states/state/ill.py [hunk 7]
```diff
@@ -66,11 +63,11 @@ def _process_html(self):
...
     def extract_docket(self, case_citation):
-        # RegEx: "{YYYY: year_4_digit} IL {docket_number}"
+        # RegEx: "{YYYY: year_4_digit} IL {docket_number}[|-U|-B|-C|-UB|-UC]"
         search = re.search(r"(?:[0-9]{4}\s+IL\s+)([0-9]+)", case_citation)
         if search:
             docket = search.group(1)
         else:
             docket = ""
-            logger.warning(f"Docket not found: {case_citation}")
+            logger.critical(f"Docket not found: {case_citation}")
```
[the remark this answers is not in the document]
Comment (satsuki-chan) [reply juriscraper/opinions/united_states/state/ill.py:72]: Could you elaborate on your question?
The endings -U, -UB, -UC, etc. aren't edited or removed from the citation, those are kept as they are listed in the court's page. However, the _**docket numbers are extracted from the citation**_, acording to the existing extracting rules for each type of court.

### For the **Supreme Court**
Extracted documents from the **Supreme Court** in CourtListener:
https://www.courtlistener.com/?q=&type=o&order_by=dateFiled%20desc&stat_Precedential=on&stat_Non-Precedential=on&court=ill
Screenshot from 2021-11-08 16-00-34

The docket numbers are the last digits from the citation. It's structure is: `{YYYY: year_4_digit} IL {docket_number}`
Example: Citation: **2020 IL 124472** (as is scraped), Docket: **124472**
https://www.illinoiscourts.gov/resources/6839236d-1943-47a2-8357-26451a3461bb/file
Screenshot from 2021-11-08 15-09-28

### For the **Appellate Court**
Extracted documents from the **Appellate Court** in CourtListener:
https://www.courtlistener.com/?q=&type=o&order_by=dateFiled%20desc&stat_Precedential=on&stat_Non-Precedential=on&court=illappct
Screenshot from 2021-11-08 16-02-02

The docket number for opinions/rulings is a composition of the appellate court district and the last digits from the citation.
Citation structure: `[{YYYY: year_4_digit}|] IL [App|] ({d: appellate district}[st|d|th]) {docket_number}[|-U|-UB|-UC|-B|-C]`
Docket structure: `{district}-{docket[0:2]}-{docket[2:]}`
Citation: **2021 IL (4th) 210157** (as is scraped), Docket: **4-21-0157**
https://ilcourtsaudio.blob.core.windows.net/antilles-resources/resources/2f934c8e-7dfa-4fac-b41b-0a1fde5af694/Dustman%20v.%20Advocate%20Aurora%20Health,%20Inc.,%202021%20IL%20(4th)%20210157.pdf
Screenshot from 2021-11-08 15-48-06

Screenshot from 2021-11-08 15-32-20

Citation: **2021 IL App (3d) 130190-C** (as is scraped), Docket: **3-13-0190**
https://ilcourtsaudio.blob.core.windows.net/antilles-resources/resources/c904f613-3b8b-422d-ba81-e818353ece30/3130190-C.pdf
Screenshot from 2021-11-08 15-36-21

Citation: **2021 IL App (3d) 150880-B** (as is scraped), Docket: **3-15-0880**
https://ilcourtsaudio.blob.core.windows.net/antilles-resources/resources/27bf5191-c376-419d-983c-9be8b0cee1d6/People%20v.%20Pacheco,%202021%20IL%20App%20(3d)%20150880-B.pdf
Screenshot from 2021-11-08 15-45-45

Citation: **2019 IL App (3d) 190203-UC** (as is scraped), Docket: **3-19-0203**
https://www.illinoiscourts.gov/resources/175c2a4c-6938-46e2-b21a-9c8dab47f397/file
Screenshot from 2021-11-08 15-51-45


The regex takes in consideration that some listed rulings' citation have errors in the search page:
Citation: **IL App (4th) 200049-U** (as is scraped), Docket: **4-20-0049**
https://ilcourtsaudio.blob.core.windows.net/antilles-resources/resources/1c0f08b1-333b-47e0-922d-209a82e76089/People%20v.%20Dishman%202021%20IL%20App%20(4th)%20200049-U.pdf
Screenshot from 2021-11-08 15-23-12

Screenshot from 2021-11-08 15-25-26

File: juriscraper/opinions/united_states/state/ill.py [hunk 8]
```diff
@@ -39,21 +37,20 @@ def _process_html(self):
...
+                # '-U' or '-U[X: a char]' are Unpublished
+                if citation[-2:] == "-U" or citation[-3:-1] == "-U":
+                    status = "Unpublished"
                 else:
-                    decision_type = get_row_column_text(row, 5)
-                    if decision_type == "Rule 23":
-                        status = "Unpublished"
-                    else:
-                        status = "Published"
+                    status = "Published"
```
[the remark this answers is not in the document]
Comment (satsuki-chan) [reply juriscraper/opinions/united_states/state/ill.py]: Done

File: juriscraper/opinions/united_states/state/ill.py (hunk 4 above)
Comment (satsuki-chan) [reply juriscraper/opinions/united_states/state/ill.py]: Done

File: juriscraper/opinions/united_states/state/illappct.py (hunk 5 above)
Comment (satsuki-chan) [reply juriscraper/opinions/united_states/state/illappct.py]: Done

**Review by flooie — APPROVED** [review]
@satsuki-chan 

Great job.  I tweaked a few things but I think you were on a great path.

**Comment by mlissner** [comment]
Whew. One down. :)

**Pull request merged**